jaxrs/WFLY-13122_EAP7-1494_Add_Json_Merge_Patch_support.adoc

```diff
@@ -0,0 +1,123 @@
+---
+categories:
+- jaxrs
+stability-level: default
+issue: https://github.com/wildfly/wildfly-proposals/issues/664
+feature-team:
+ developer: ema
+ sme:
+  - ema
+ outside-perspective:
+  -
+promotes:
+promoted-by:
+---
+= Add Json Merge Patch support
+:author:            Jim Ma
+:email:             ema@redhat.com
+:toc:               left
+:icons:             font
+:keywords:          json merge patch,resteasy,jaxrs
+:idprefix:
+:idseparator:       -
+:issue-base-url:    https://issues.jboss.org/browse
+
+== Overview
+This feature is to support JSON Merge Patch spec as described in: https://tools.ietf.org/html/rfc7386
+With Json merge patch support, the client can send a json http request with Content-Type "application/merge-patch+json",
+and the json merge request will be directly apply to the target resource with the merge function as spec described.
+
+Resteasy provides two implementation for JSON merge patch. One based on jsonp(abbreviation jsonp will be used in this page) and another one is based on 
+json-patch and jackson2(abbreviation jackson2 will be used in this page). By default, Resteasy uses jsonp implementation, while the jackson2 implementation is only enabled 
+when the context parameter `resteasy.patchfilter.legacy=true` is set.
```

Comment:
We should only discuss the Jakarta JSON Processing option IMO. We shouldn't mention the legacy option.

Reply:
This is requested by @honza-kasik, as this is still there and still can be set to switch.